code-challenges.test.js

```diff
@@ -14,32 +14,128 @@
 
+/*
+First thing that comes to mind is a conditional statement per each letter. I also think a for loop so it can iterate through the strings characters and identify them. I think there should be other ways but I might come back and see if I can do better. 
+*/
+
+const toNum = (string) => {
+    const newArray = []
+    for(let i = 0; i < string.length; i++){
+    if (string[i] == "a"){
+        newArray.push(string[i].replace("a", 4))
+    }
+    else if (string[i] == "e"){
+        newArray.push(string[i].replace("e", 3))
+    }
+    else if (string[i] == "E"){
+        newArray.push(string[i].replace("E", 3))
+    }
+    else if (string[i] == "i"){
+        newArray.push(string[i].replace("i", 1))
+    }
+    else if (string[i] == "o"){
+        newArray.push(string[i].replace("o", 0))
+    }
+    else {newArray.push(string[i])}
+}
+    return newArray.join("")
+}
+
+
+
+
+
+
+
 // --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
+/*
+Input: fruitArray, letterA
+Output: ["Mango", "Apricot", "Peach"]
+Input: fruitArray, letterE
+Output: ["Cherry", "Blueberry", "Peach"]
+Starting with a describe statement, I will call it "containing". Can't really think of anything else.
+For my it statement I will pass down the prompt in plain English in a string.
+For the it statement function I will have 2 expect statements, one per variable
+For my matcher I will use .toEqual as we want the exact value in the expected output
+Test output will be expected output per variable 
+*/
+
 const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
 
 const letterA = "a"
 // Expected output: ["Mango", "Apricot", "Peach"]
 const letterE = "e"
 // Expected output: ["Cherry", "Blueberry", "Peach"]
 
+describe("containing", () => {
+    it("function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.", () => {
+      expect(containing(fruitArray, letterA)).toEqual(["Mango", "Apricot", "Peach"])
+      expect(containing(fruitArray, letterE)).toEqual(["Cherry", "Blueberry", "Peach"]) 
+    })
+  })
+
+ // ReferenceError: containing is not defined
+
+
 // b) Create the function that makes the test pass.
 
+/* first, we need to account for any letters that may be upper or lower case. We can lowercase them with .toLowerCase method. We have to map through so we can look through the array. 
+*/
+
+const containing = (array, letter) => {
+    let newArray = array.map(value => value.toLowerCase())
+   return newArray.filter(value => value.includes(letter)).map(value => value.charAt(0).toUpperCase() + value.slice(1))
+}
+
+/* I used .include method so it will see if it includes that letter. .charAt to capitalize first letter like expected output. 
```

Comment:
There is a way to simplify this problem down by removing the maps and using .toLowerCase within your filter.